### PR TITLE
Deal with some occurrences of JRequest

### PR DESCRIPTION
**Pull request by realityking**
Not quite finished, but I wanted to get out what I had.

After this is merged only the following classes would still use JRequest:

JClientHelper
JSession
JControllerAdmin
JControllerForm
JModuleHelper

Also the tests for JDocumentRendererAtom and JDocumentRendererRSS also still use JRequest.

### CODE REVIEW

File: libraries/legacy/application/helper.php [hunk 1]
```diff
@@ -52,14 +52,15 @@ public static function getComponentName($default = null)
 			return $option;
 		}
 
-		$option = strtolower(JRequest::getCmd('option'));
+		$input = JFactory::getApplication();
```
Comment (chdemko) [inline libraries/legacy/application/helper.php]: Do you forget

```
JFactory::getApplication()->input
```

?

File: libraries/legacy/application/helper.php [hunk 2]
```diff
@@ -52,14 +52,15 @@ public static function getComponentName($default = null)
...
+		$option = strtolower($input->get('option'));
 
 		if (empty($option))
 		{
 			$option = $default;
 		}
 
-		JRequest::setVar('option', $option);
+		$input->get('option', $option);
```
Comment (chdemko) [inline libraries/legacy/application/helper.php]: Do you mean

```
$input->set('option', $option);
```

?

File: libraries/legacy/controller/admin.php [hunk 3]
```diff
@@ -211,8 +213,8 @@ public function publish()
 				$this->setMessage(JText::plural($ntext, count($cid)));
 			}
 		}
-		$extension = JRequest::getCmd('extension');
-		$extensionURL = ($extension) ? '&extension=' . JRequest::getCmd('extension') : '';
+		$extension = $input->get('extension');
```
Comment (chdemko) [inline libraries/legacy/controller/admin.php:216]: Do you mean

```
$extension = $input->getCmd('extension');
```

File: libraries/legacy/controller/admin.php [hunk 4]
```diff
@@ -211,8 +213,8 @@ public function publish()
...
+		$extensionURL = ($extension) ? '&extension=' . $input->get('extension') : '';
```
Comment (chdemko) [inline libraries/legacy/controller/admin.php]: idem

File: libraries/legacy/controller/controller.php [hunk 5]
```diff
@@ -614,8 +616,9 @@ public function display($cachable = false, $urlparams = array())
 	{
 		$document = JFactory::getDocument();
 		$viewType = $document->getType();
-		$viewName = JRequest::getCmd('view', $this->default_view);
-		$viewLayout = JRequest::getCmd('layout', 'default');
+		$input    = JFactory::getApplication()->input;
+		$viewName = $input->get('view', $this->default_view);
```
Comment (chdemko) [inline libraries/legacy/controller/controller.php:620]: Do you mean

```
$viewName = $input->getCmd('view', $this->default_view);
```

File: libraries/legacy/controller/controller.php [hunk 6]
```diff
@@ -614,8 +616,9 @@ public function display($cachable = false, $urlparams = array())
...
+		$viewLayout = $input->get('layout', 'default');
```
Comment (chdemko) [inline libraries/legacy/controller/controller.php:621]: idem

File: libraries/legacy/controller/controller.php [hunk 7]
```diff
@@ -633,7 +636,7 @@ public function display($cachable = false, $urlparams = array())
 		// Display the view
 		if ($cachable && $viewType != 'feed' && $conf->get('caching') >= 1)
 		{
-			$option = JRequest::getCmd('option');
+			$option = $input->get('layout');
```
Comment (chdemko) [inline libraries/legacy/controller/controller.php:639]: Do you mean

```
$option = $input->getCmd('layout');
```

File: libraries/legacy/controller/form.php [hunk 8]
```diff
@@ -484,8 +485,9 @@ public function getModel($name = '', $prefix = '', $config = array('ignore_reque
 	 */
 	protected function getRedirectToItemAppend($recordId = null, $urlVar = 'id')
 	{
-		$tmpl = JRequest::getCmd('tmpl');
-		$layout = JRequest::getCmd('layout', 'edit');
+		$input  = JFactory::getApplication()->input;
+		$tmpl   = $input->get('tmpl');
```
Comment (chdemko) [inline libraries/legacy/controller/form.php:489]: Do you mean

```
$tmpl   = $input->getCmd('tmpl');
```

File: libraries/legacy/controller/form.php [hunk 9]
```diff
@@ -484,8 +485,9 @@ public function getModel($name = '', $prefix = '', $config = array('ignore_reque
...
+		$layout = $input->get('layout', 'edit');
```
Comment (chdemko) [inline libraries/legacy/controller/form.php:490]: idem

File: libraries/legacy/controller/form.php [hunk 10]
```diff
@@ -516,7 +518,7 @@ protected function getRedirectToItemAppend($recordId = null, $urlVar = 'id')
 	 */
 	protected function getRedirectToListAppend()
 	{
-		$tmpl = JRequest::getCmd('tmpl');
+		$tmpl = JFactory::getApplication()->input->get('tmpl');
```
Comment (chdemko) [inline libraries/legacy/controller/form.php:521]: Do you mean

```
$tmpl = JFactory::getApplication()->input->getCmd('tmpl');
```

File: libraries/legacy/controller/form.php [hunk 11]
```diff
@@ -559,11 +561,12 @@ public function save($key = null, $urlVar = null)
...
 		// Initialise variables.
-		$app = JFactory::getApplication();
-		$lang = JFactory::getLanguage();
+		$app   = JFactory::getApplication();
+		$input = $app->input;
+		$lang  = JFactory::getLanguage();
 		$model = $this->getModel();
 		$table = $model->getTable();
-		$data = JRequest::getVar('jform', array(), 'post', 'array');
+		$data  = JRequest::getVar('jform', array(), 'post', 'array');
```
Comment (chdemko) [inline libraries/legacy/controller/form.php:569]: Why don't you use $input ?

File: libraries/legacy/form/field/media.php [hunk 12]
```diff
@@ -50,13 +50,12 @@ protected function getInput()
 		$asset = $this->form->getValue($assetField) ? $this->form->getValue($assetField) : (string) $this->element['asset_id'];
 		if ($asset == '')
 		{
-			$asset = JRequest::getCmd('option');
+			$asset = JFactory::getApplication()->input->get('option');
```
Comment (chdemko) [inline libraries/legacy/form/field/media.php:53]: Do you mean

```
$asset = JFactory::getApplication()->input->getCmd('option');
```

File: libraries/legacy/model/model.php [hunk 13]
```diff
@@ -496,7 +496,7 @@ protected function cleanCache($group = null, $client_id = 0)
 		$dispatcher = JDispatcher::getInstance();
 
 		$options = array(
-			'defaultgroup' => ($group) ? $group : (isset($this->option) ? $this->option : JRequest::getCmd('option')),
+			'defaultgroup' => ($group) ? $group : (isset($this->option) ? $this->option : JFactory::getApplication()->input->get('option')),
```
Comment (chdemko) [inline libraries/legacy/model/model.php:499]: Do you mean

    'defaultgroup' => ($group) ? $group : (isset($this->option) ? $this->option : JFactory::getApplication()->input->getCmd('option')),

File: libraries/legacy/application/helper.php (hunk 1 above)
Comment (AmyStephen) [inline libraries/legacy/application/helper.php]: JFactory::getApplication()->input->getCmd('option') even.

Comment (realityking) [inline libraries/legacy/application/helper.php]: Done.

File: libraries/legacy/application/helper.php (hunk 2 above)
Comment (realityking) [inline libraries/legacy/application/helper.php]: Done.

File: libraries/legacy/controller/admin.php (hunk 3 above)
Comment (realityking) [inline libraries/legacy/controller/admin.php:216]: cmd is the default filter so we don't have to add it explicitly.

File: libraries/legacy/controller/admin.php (hunk 4 above)
Comment (realityking) [inline libraries/legacy/controller/admin.php]: Not sure what ideam means but I think I fixed what you mean.

File: libraries/legacy/controller/controller.php (hunk 5 above)
Comment (realityking) [inline libraries/legacy/controller/controller.php:620]: Same as above, cmd is the default filter.

File: libraries/legacy/controller/controller.php (hunk 7 above)
Comment (realityking) [inline libraries/legacy/controller/controller.php:639]: Same as above, cmd is the default filter.

File: libraries/legacy/controller/controller.php (hunk 6 above)
Comment (realityking) [inline libraries/legacy/controller/controller.php:621]: see above

File: libraries/legacy/controller/form.php (hunk 11 above)
Comment (realityking) [inline libraries/legacy/controller/form.php:569]: I'll fix those up in the next batch.

**Comment by realityking** [comment]
Thanks for the comments! getCmd() isn't really needed since cmd is the default filter, I opted for the shortest syntax possible while preferring get() over the magic methods.

**Comment by elinw** [comment]
Thanks for doing these Rouven. Helps a lot.

**Pull request merged**